apps-rendering/src/adSlot.tsx

```diff
@@ -0,0 +1,131 @@
+import type { SerializedStyles } from '@emotion/react';
+import { css, ThemeProvider } from '@emotion/react';
+import {
+	background,
+	text,
+} from '@guardian/common-rendering/src/editorialPalette';
+import type { ArticleFormat } from '@guardian/libs';
+import {
+	from,
+	headline,
+	remSpace,
+	textSans,
+	until,
+} from '@guardian/source-foundations';
+import { Button, buttonThemeBrandAlt } from '@guardian/source-react-components';
+import type { FC, ReactElement } from 'react';
+import { darkModeCss, wideContentWidth } from 'styles';
+
+type Props = {
+	className: string;
+	paragraph: number;
+	format: ArticleFormat;
+};
+
+const adLabelsStyles = (format: ArticleFormat): SerializedStyles => css`
+	${textSans.xsmall()}
+	color: ${text.adLabel(format)};
+	padding: ${remSpace[3]};
+	float: left;
+	width: calc(100% - ${remSpace[3]} - ${remSpace[3]});
+
+	h1 {
+		margin: 0;
+		float: left;
+		font-size: 16px;
+		font-weight: 400;
+
+		${darkModeCss`
+		color: ${text.adLabelDark(format)};
+	`}
+	}
+`;
+
+const supportBannerStyles = (format: ArticleFormat): SerializedStyles => css`
+	padding: ${remSpace[3]};
+	background-color: ${background.supportBanner(format)};
+
+	h1 {
+		${headline.xxxsmall()};
+		margin-top: 0;
+	}
+
+	button {
+		margin-top: ${remSpace[3]};
+	}
+
+	${darkModeCss`
+		background-color: ${background.supportBannerDark(format)};
+	`}
+`;
+
+const styles = (format: ArticleFormat): SerializedStyles => css`
+	clear: both;
+	margin: ${remSpace[4]} 0;
+	color: ${text.adSlot(format)};
+	background: ${background.adSlot(format)};
+
+	&.hidden {
+		display: none;
+	}
+
+	${darkModeCss`
+		background-color: ${background.adSlotDark(format)};
+	`}
+
+	${from.desktop} {
+		position: absolute;
+		margin-left: calc(${wideContentWidth}px + ${remSpace[4]});
+		min-width: 300px;
+		margin-bottom: ${remSpace[6]};
+	}
+
+	${until.phablet} {
+		margin: 1em -${remSpace[3]};
+	}
+
+	// This class is applied in the nativeCommunication module
+	// to the first ad slot if Teads ads are enabled.
+	&.ad-slot-square {
```

Comment:
This is a bill confusing as the ad slot has two ways of having a defined height. Either on the wrapper, as here, or on the actual slot, as below.

Could this be unified?

Reply:
Well spotted! This should actually live in the actual ad slot rather than the wrapper. I've moved it :)